### PR TITLE
Add exceptions doc for HttpHeaders.Add Method

### LINKED ISSUE #1016: "throws" omitted
Remarks section makes notice of validation of value but I can't find information of what happens when validation fails.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: acef50e2-0c59-cfb6-0d3c-99c135ca76cb
* Version Independent ID: 11531e0e-dca1-e0c0-3b47-f8b04a70eaad
* Content: [HttpHeaders.Add Method (System.Net.Http.Headers)](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.headers.httpheaders.add?f1url=https%3A%2F%2Fmsdn.microsoft.com%2Fquery%2Fdev15.query%3FappId%3DDev15IDEF1%26l%3DEN-US%26k%3Dk(System.Net.Http.Headers.HttpHeaders.Add);k(SolutionItemsProject);k(TargetFrameworkMoniker-.NETFramework,Version%3Dv4.5);k(DevLang-csharp)%26rd%3Dtrue%26f%3D255%26MSPPError%3D-2147217396&view=netframework-4.7.2)
* Content Source: [xml/System.Net.Http.Headers/HttpHeaders.xml](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System.Net.Http.Headers/HttpHeaders.xml)
* Product: **dotnet-api**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

### PR DESCRIPTION
**Pull request by caesar-chen**
Fixes #1016

cc: @karelz @davidsh

### CODE REVIEW

**Comment by davidsh** [comment]
@rpetrusha

File: xml/System.Net.Http.Headers/HttpHeaders.xml [hunk 1]
```diff
@@ -113,6 +113,11 @@
   
  ]]></format>
         </remarks>
+        <exception cref="T:System.ArgumentException">The name cannot be null or empty.</exception>
+        <exception cref="T:System.ArgumentNullException">The values cannot be null or empty.</exception>
+        <exception cref="T:System.InvalidOperationException">Misused header name. Make sure request headers are used with HttpRequestMessage, response headers with HttpResponseMessage, and content headers with HttpContent objects.</exception>
+        <exception cref="T:System.FormatException">The header name format is invalid.</exception>
```
Comment (rpetrusha) [inline xml/System.Net.Http.Headers/HttpHeaders.xml]: Typically, an exception type is listed once, with multiple exception conditions. My suggested correction for line 119 will do that, and line 120 should be deleted.

Comment (rpetrusha) [inline xml/System.Net.Http.Headers/HttpHeaders.xml]:
```suggestion
        <exception cref="T:System.FormatException">The header name format is invalid.

-or-

New line characters in header values must be followed by a white-space character. 
        </exception>
```

File: xml/System.Net.Http.Headers/HttpHeaders.xml [hunk 2]
```diff
@@ -158,6 +163,10 @@
   
  ]]></format>
         </remarks>
+        <exception cref="T:System.ArgumentException">The name cannot be null or empty.</exception>
+        <exception cref="T:System.InvalidOperationException">Misused header name. Make sure request headers are used with HttpRequestMessage, response headers with HttpResponseMessage, and content headers with HttpContent objects.</exception>
+        <exception cref="T:System.FormatException">The header name format is invalid.</exception>
```
Comment (rpetrusha) [inline xml/System.Net.Http.Headers/HttpHeaders.xml]: The same fix applies here.

**Comment by caesar1995** [comment]
@rpetrusha Thanks, fixed in the new commit.

**Comment by rpetrusha** [comment]
Thanks for the additional changes, @caesar1995. I'll merge your PR now.

**Pull request merged**